ArchUnitNET/Domain/PlantUml/Export/PlantUmlDependency.cs

```diff
@@ -17,15 +18,122 @@ public PlantUmlDependency(string origin, string target, DependencyType dependenc
             DependencyType = dependencyType;
         }
 
-        public string GetPlantUmlString(RenderOptions renderOptions)
+        private static int CountOfDots(string str)
+        {
+            return str.Count(c => c == '.');
+        }
+
+        public int OriginCountOfDots()
+        {
+            return CountOfDots(Origin);
+        }
+
+        public int TargetCountOfDots()
+        {
+            return CountOfDots(Target);
+        }
+
+        public string GetPlantUmlString(RenderOptions renderOptions = null)
         {
             switch (DependencyType)
             {
                 case DependencyType.OneToOne:
-                    return "\"" + Origin + "\"" + " --|> " + "\"" + Target + "\"" + Environment.NewLine;
+                    return "[" + Origin + "]" + " --|> " + "[" + Target + "]" + Environment.NewLine;
+
                 case DependencyType.OneToMany:
-                    return "\"" + Origin + "\"" + " \"1\" --|> \"many\" " + "\"" + Target + "\"" + " " +
+                    return "[" + Origin + "]" + " \"1\" --|> \"many\" " + "[" + Target + "]" + " " +
                            Environment.NewLine;
+
+                case DependencyType.OneToOneIfSimilarNamespace:
```

Comment:
I think there is an opportunity here to extract code to a couple of functions :)